src/components/QuizDisplay.tsx

```diff
@@ -1,112 +1,76 @@
-import { useRef, useState } from "react";
+import { useState } from "react";
 import SingleQuestion from "./SingleQuestion";
-import { IAnswerInfo, IQuestionInfo, QuizStatusType } from "../types";
-import Report from "./Report";
-
-interface QuizDisplayProps {
-  allQuestions: IQuestionInfo[];
-  quizStatus: QuizStatusType;
-  setQuizStatus: React.Dispatch<React.SetStateAction<QuizStatusType>>;
-}
+import { IQuizDisplayProps, IAnswerInfo } from "../types";
+import ReturnToQuizSetupBtn from "./ReturnToQuizSetupBtn";
 
 const QuizDisplay = ({
   allQuestions,
-  quizStatus,
   setQuizStatus,
-}: QuizDisplayProps) => {
+  noOfCorrectAnswers,
+  handleReturnToQuizSetup,
+}: IQuizDisplayProps) => {
   const [selectedAnswer, setSelectedAnswer] = useState<number | null>(null);
   const [allAnswers, setAllAnswers] = useState<IAnswerInfo[]>([]);
   const [answerIsSubmitted, setAnswerIsSubmitted] = useState<boolean>(false);
-  const [nextButtonIsDisabled, setNextButtonIsDisabled] =
-    useState<boolean>(true);
+  const [isNextBtnDisabled, setIsNextBtnDisabled] = useState<boolean>(true);
   const [currentQNo, setCurrentQNo] = useState<number>(0);
 
-  const correctAnswers = useRef(0);
-
   const handleSubmitAnswer = () => {
     setAnswerIsSubmitted(true);
-    setNextButtonIsDisabled(false);
-    if (allAnswers[selectedAnswer!].text === correctAnswer)
-      correctAnswers.current++;
+    setIsNextBtnDisabled(false);
+    if (allAnswers[selectedAnswer!].text === correctAnswer.text)
+      noOfCorrectAnswers.current++;
   };
 
   const handleClickNext = () => {
     if (currentQNo < allQuestions.length - 1) {
       setCurrentQNo((prev) => prev + 1);
       setSelectedAnswer(null);
       setAnswerIsSubmitted(false);
-      setNextButtonIsDisabled(true);
+      setIsNextBtnDisabled(true);
     } else {
       setQuizStatus("completed");
     }
   };
 
-  const correctAnswer = quizStatus
-    ? allQuestions[currentQNo].correct_answer
-    : null;
+  const correctAnswer = allAnswers.filter(
+    (answerInfo) => answerInfo.isCorrect === true
+  )[0];
 
-  const correctStyle = {
-    color: "green",
-    display: answerIsSubmitted ? "block" : "none",
-  };
-
-  const incorrectStyle = {
-    color: "red",
-    display: answerIsSubmitted ? "block" : "none",
-  };
+  const isSubmitBtnDisabled =
+    selectedAnswer == null || answerIsSubmitted === true;
 
```

Comment:
I like it, I've made this change

Reply:
I've had to change this back to the following
const isSubmitBtnDisabled = selectedAnswer == null || isAnswerSubmitted;

When I changed it to your suggestion, it did not allow me to submit the first answer from the list of answers - selectedAnswer is of type number and it's the index of the answer from the list of answers. So when the first answer is clicked, the 'selectedAnswer' state becomes 0 which means !selectedAnswer evaluates to true and the submit button stays disabled which isn't what I want.

I can either leave this line with selectedAnswer == null or I can go back to your suggestion but if I change it back to your suggestion, I need to change the way selectedAnswer is defined so it ranges from 1-4 instead of 0-3 (or 1-2 instead of 0-1 when its true or false questions).

I think it'll be better if I leave it as
const isSubmitBtnDisabled = selectedAnswer == null || isAnswerSubmitted;
so that I don't need to do any further refactoring but what do you think?